### PR TITLE
Add thiscall calling convention, mostly for hooking purpose

### PR DESCRIPTION
**Pull request by ba0f3**

### CODE REVIEW

**Comment by Araq** [comment]
Needs a changelog entry and to be mentioned in the documentation.

File: changelog.md [hunk 1]
```diff
@@ -159,6 +159,8 @@ proc mydiv(a, b): int {.raises: [].} =
   The reason for this is that `DivByZeroDefect` inherits from `Defect` and
   with `--panics:on` `Defects` become unrecoverable errors.
 
+- Added `thiscall` calling convention as specified by Microsoft, mostly for hooking purpose
```
Comment (Araq) [inline changelog.md:162]: What's a "hooking purpose"?

Comment (ba0f3) [reply changelog.md:162]: kind of detours/api hooking/patching. cuz I dont see a use case of `thiscall` in real life for Nim. Should I remove that term?

Comment (Araq) [reply changelog.md:162]: It's ok.

**Pull request merged**